rust/ranked-prolly-tree/README.md

```diff
@@ -0,0 +1,153 @@
+# ranked-prolly-tree
+
+
+`ranked-prolly-tree` (RPT) is a key-value store implemented as a [prolly tree], utilizing a flexible content-addressed block storage backend. RPT is designed to be the foundation of a lazy database, utilizing partial sync on demand.
+
+## Usage
+
+```rust
+use ranked_prolly_tree::{Tree, Result};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let mut tree = Tree::<64, _>::default();
+    let hello = String::from("hello").into_bytes();
+    let world = String::from("world").into_bytes();
```

Comment:
Same could be said about the values. If those are just bytes you loose ability to swap encoding, but if they are more higher level data that would allow pluggable encoding which I think would be of high value.

Reply:
This is less clear to me than on the key side (https://github.com/commontoolsinc/system/issues/272); encoding is pluggable, and only requires inputs implementing serde's `Serialize` and `DeserializeOwned` but still wouldn't support changing encodings on the fly (though a migrator is possible).